src/components/Board.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState } from 'react'
+import Square from './Square'
+import { checkWinner, checkStale } from '../utils'
+import Confetti from 'react-dom-confetti'
+import confettiConfig from './confetti-config'
+import Restart from './Restart'
+
+//component, destructured props of type BoardTypeProps
+const Board = ({
+  gameHistory,
+  addGameHistory,
+  currentTurn,
+}: BoardTypeProps) => {
+  const [gameState, setGameState] = useState<GameState>(gameHistory[0])
+  const [isWinner, setIsWinner] = useState<boolean>(false)
+  const [isStale, setIsStale] = useState<boolean>(false)
+  const [winningSquares, setWinningSquares] = useState<number[]>([])
+  const [sliderValue, setSliderValue] = useState<number>(0)
+
+  /**
+   *
+   * @param gameState
+   * @param squareIndex
+   * @param newValue
+   * @returns
+   */
+  const updateBoardState = (
+    gameState: GameState,
+    squareIndex: number,
+    newValue: string
+  ) => {
+    const newGameState = gameState.map((square, index) => {
+      if (squareIndex === index) return newValue
+      return square
+    })
+
+    return newGameState
+  }
+
+  /**
+   * Handles updating the sate of the board string array
+   * given a value from a square component
+   * @param id
+   * @param value
+   */
+  const onSquareClick = (id: number, value: string) => {
+    const newGameState = updateBoardState(gameState, id, value)
+    const possibleWinningSquares = checkWinner(newGameState)
```

Comment:
```suggestion
    const isGameWon = checkWinner(newGameState)
```

any time a function returns a boolean, I'd always prefix the return value with is- or has-